src/Microsoft.PowerShell.Commands.Diagnostics/GetEventCommand.cs

```diff
@@ -1538,6 +1542,41 @@ private string HandleDataHashValue(Object value)
 
 
         //
+        // HandleNamedDataHashValue helper for hashtable structured query builder.
+        // Constructs and returns named event data field XPath portion as a string.
+        // Fix Issue #2327
+        //
+        private string HandleNamedDataHashValue(String key, Object value)
+        {
+            string ret = "";
+            if (value is Array)
```

Comment:
This is doing two casts, once with the is keyword and again with the explicit cast.  We normally use this pattern to avoid the second cast:
Array dataArray = value as Array;
if (dataArray != null)
{
...